Explorer/Assets/DCL/Interaction/Systems/ProcessPointerEventsSystem.cs

```diff
@@ -0,0 +1,238 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.Input;
+using DCL.Interaction.PlayerOriginated.Components;
+using DCL.Interaction.PlayerOriginated.Systems;
+using DCL.Interaction.PlayerOriginated.Utility;
+using DCL.Interaction.Utility;
+using ECS.Abstract;
+using ECS.Unity.PrimitiveRenderer.Components;
+using ECS.Unity.Transforms.Components;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using UnityEngine.Pool;
+
+namespace DCL.Interaction.Systems
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [UpdateAfter(typeof(PlayerOriginatedRaycastSystem))]
+    [LogCategory(ReportCategory.INPUT)]
+    public partial class ProcessPointerEventsSystem : BaseUnityLoopSystem
+    {
+        private readonly IEntityCollidersGlobalCache entityCollidersGlobalCache;
+        private readonly Material hoverMaterial;
+        private readonly Material hoverOorMaterial;
+        private readonly IEventSystem eventSystem;
+        private readonly IReadOnlyDictionary<InputAction, UnityEngine.InputSystem.InputAction> sdkInputActionsMap;
+        private readonly Dictionary<EntityReference, Dictionary<EntityReference, Material[]>> originalMaterialsByEntity = new ();
```

Comment:
🔴 You can't cache by `EntityReference` as they exist only in the context of the particular world: they don't contain a reference to the world itself and the same content can be present in different worlds.
It's another argument for moving this logic to the scene world